### PR TITLE
LPS-184639

### PR DESCRIPTION
**Pull request by ces-quanhuynh**
Hi team,
Please help me to review this PR
Thank you.
Quan

# Motivation
[LPS-184639](https://liferay.atlassian.net/browse/LPS-184639) [BE] Add a new table to map each folder id with its propagation settings

# Solution
For the first task, I created the new table with these column suggesting by Product team ( companyId, groupId, classNameId, classPK, propagation ) and propagationPermissionId as PK. 
I decide to create it in portal-impl because it should be generic and will be used by WC, KB,....

cc @locpham97 @ces-luannguyen @tototrinh

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/service.xml [hunk 1]
```diff
@@ -1762,6 +1762,40 @@
 			<finder-column name="smallValue" />
 		</finder>
 	</entity>
+	<entity change-tracking-enabled="true" local-service="true" name="PropagationPermission" remote-service="true">
```
Comment (tototrinh) [inline portal-impl/src/com/liferay/portal/service.xml:1765]: The table name suggested by the product team is`PermissionPropagationEntry`?

**Comment by ces-locpham** [comment]
I think it is better to move all packageinfo files to the baseline commit.

**Comment by ces-locpham** [comment]
Change the second commit to buildService

File: portal-impl/src/com/liferay/portal/service.xml (hunk 1 above)
Comment (ces-quanhuynh) [reply portal-impl/src/com/liferay/portal/service.xml]: I've updated the code.
Thank you!

**Comment by ces-quanhuynh** [comment]
Hi @locpham97 @tototrinh,
I've updated the code following your reviews.
Please take a look.
Thanks,
Quan

**Review by ces-locpham — APPROVED** [review]

**Review by ces-totrinh — APPROVED** [review]

**Comment by ces-totrinh** [comment]
Hi @hhtank, We have changed the solution for [LPS-154515](https://liferay.atlassian.net/browse/LPS-154515). Instead of adding a new column to DLFolder, we will add a new table to map each folder id with its propagation settings. Please see [the discussion on Slack](https://liferay.slack.com/archives/C054ZCFJBGT/p1686563551146009) for more detail.
This PR passed our review, please help take a look.
Thanks.

**Comment by holatuwol** [comment]
For creating a new entity, you would likely add a SQL script like this, which added a `CountryLocalization` table:
https://github.com/liferay/liferay-portal/blob/7.4.3.60-ga60/sql/update-7.3.0-7.4.0-country.sql

Then you'd run it the same way we do in `UpgradeCountry`:
https://github.com/liferay/liferay-portal/blob/7.4.3.60-ga60/portal-impl/src/com/liferay/portal/upgrade/v7_4_x/UpgradeCountry.java#L37

**Comment by ces-locpham** [comment]
Please baseline @ces-quanhuynh

**Comment by ces-quanhuynh** [comment]
I've added baseline commit.
Many thanks @locpham97!

File: portal-impl/src/com/liferay/portal/service.xml [hunk 2]
```diff
@@ -1762,6 +1762,26 @@
...
+		<column name="permissionPropagationEntryId" primary="true" type="long" />
+
+		<!-- Audit fields -->
+
+		<column name="companyId" type="long" />
+
+		<!-- Group instance -->
+
+		<column name="groupId" type="long" />
+
```
Comment (locpham97) [inline portal-impl/src/com/liferay/portal/service.xml:1778]: Change the Group Instance first and Audit fields second

**Comment by ces-totrinh** [comment]
@ces-quanhuynh please SF. Thanks.

**Comment by ces-quanhuynh** [comment]
> @ces-quanhuynh please SF. Thanks.

@tototrinh Thanks. I've done the SF.

File: portal-impl/src/com/liferay/portal/service.xml [hunk 3]
```diff
@@ -1423,6 +1423,26 @@
...
+		<!-- Audit fields -->
+
+		<column name="companyId" type="long" />
+
+		<!-- Other fields -->
+
+		<column name="classPK" type="long" />
+		<column name="classNameId" type="long" />
```
Comment (tototrinh) [inline portal-impl/src/com/liferay/portal/service.xml:1443]: Reorder:
```
		<column name="classNameId" type="long" />
		<column name="classPK" type="long" />
```

Comment (ces-quanhuynh) [reply portal-impl/src/com/liferay/portal/service.xml]: Thanks @tototrinh
I've reordered them.

**Review by ces-locpham — APPROVED** [review]

**Review by ces-totrinh — APPROVED** [review]
LGTM. Thanks.

**Comment by ces-totrinh** [comment]
@hhtank, we finished the internal review. Please help take a look and give your feedback. Thanks.

**Comment by hhtank** [comment]
@tototrinh
LGTM, thanks!

**Comment by ces-totrinh** [comment]
Merged. Thanks all.

**Pull request merged**